src/incubator/TextField/useFieldState.ts

```diff
@@ -68,9 +69,13 @@ export default function useFieldState({
     setIsValid(_isValid);
     setFailingValidatorIndex(_failingValidatorIndex);
 
+    if (!_isValid && _failingValidatorIndex !== undefined && _failingValidatorIndex >= 0) {
```

Comment:
Are the `_failingValidatorIndex !== undefined && _failingValidatorIndex >= 0` conditions here to make sure it's not undefined? 
if so, please use `!_.isUndefined(_failingValidatorIndex)`